src/components/sidenav/sidenav.js

```diff
@@ -278,26 +308,43 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
      * Toggle the SideNav view and attach/detach listeners
      * @param isOpen
      */
-    function updateIsOpen(isOpen) {
+    function updateIsOpen(isOpen, oldValue) {
+      if (skipNextUpdate) {
+        skipNextUpdate = false;
+        return;
+      }
+
       // Support deprecated md-sidenav-focus attribute as fallback
       var focusEl = $mdUtil.findFocusTarget(element) || $mdUtil.findFocusTarget(element,'[md-sidenav-focus]') || element;
       var parent = element.parent();
 
+      // little hack to check show abbility
+      if (!skipSidenav) {
```

Comment:
If we not skip the scope.isOpen update, I remove the `md-closed` class temporary (won't be affect the view, because it's outside of the viewport) and then I can check for a `hide` attribute.